chore: Guide Enhancements (#74)
Pycord-Development/guide: docs/getting-started/more-features.mdx

title: More Features
description: More features to make your bot cool and snazzy.
---

import {
  DiscordInteraction,
  DiscordMessage,
  DiscordMessages,
  DiscordEmbed,
  DiscordEmbedField,
  DiscordEmbedFields,

> } from "@discord-message-components/react";
> import "@discord-message-components/react/styles";

Reviewer: Update for the new package name
```suggestion
} from "discord-message-components/packages/react";
import "discord-message-components/packages/react/dist/style.css";
```

Embeds are a Discord feature that allows applications to format their messages in cool embedded format,
enabling your bot to lay out messages with a lot of text into neat fields.
 
<DiscordMessages>
    <DiscordMessage author="Guide Bot" avatar="red" bot>
        <DiscordEmbed
        embedTitle="Pycord Guide"
        url="https://guide.pycord.dev"
        authorIcon="/img/logo.png"
        authorName="Pycord Team"
        authorUrl="https://github.com/Pycord-Development/guide"
        thumbnail="/img/logo.png"
        borderColor="#5865F2"
        timestamp="01/09/2021"
        footerIcon="/img/logo.png"
        image="/img/banner-v3.png"
        >

>             Pycord Guide is a detailed guide that explains how to use Pycord and all it's features.

Reviewer 2: ```suggestion
            Pycord Guide is a detailed guide that explains how to use Pycord and all of its features.
```

> <DiscordMessages>
>     <DiscordMessage author="Guide Bot" avatar="red" bot>
>         <DiscordEmbed
>         embedTitle="Pycord Guide"
>         url="https://guide.pycord.dev"
>         authorIcon="/img/logo.png"
>         authorName="Pycord Team"
>         authorUrl="https://github.com/Pycord-Development/guide"
>         thumbnail="/img/logo.png"
>         borderColor="#5865F2"
>         timestamp="01/09/2021"
>         footerIcon="/img/logo.png"
>         image="/img/banner-v3.png"
>         >
>             Pycord Guide is a detailed guide that explains how to use Pycord and all it's features.
>             <DiscordEmbedFields slot="fields">
>                 <DiscordEmbedField fieldTitle="Getting Started">
>                     The <a href="/category/getting-started">Getting Started section</a> explains how you can get a brand new bot created and running from scratch.
>                 </DiscordEmbedField>
>                 <DiscordEmbedField fieldTitle="Interactions" inline="true">
>                     <a href="/interactions">Interactions</a> with Pycord
>                 </DiscordEmbedField>
>                 <DiscordEmbedField fieldTitle="Extensions" inline="true">
>                     Pycord's various <a href="/category/extensions">Extensions</a>
>                 </DiscordEmbedField>
>                 <DiscordEmbedField fieldTitle="Popular Topics" inline="true">
>                     <a href="/category/popular-topics">Other Popular Topics</a>
>                 </DiscordEmbedField>
>                 <DiscordEmbedField fieldTitle="And More!">
>                     We have so much more! Just explore, and you will find everything you need. If you want another page added, open an issue on the <a href="https://github.com/Pycord-Development/guide">GitHub</a>.
>                 </DiscordEmbedField>
>             </DiscordEmbedFields>
>             <span slot="footer">Created with 💖 by the Pycord Team & Contributers</span>
>         </DiscordEmbed>
>     </DiscordMessage>
> </DiscordMessages>

Reviewer: ```suggestion
<DiscordComponent>
    <DiscordMessage author="Guide Bot" avatar="red" bot>
        <DiscordEmbed
        embedTitle="Pycord Guide"
        url="https://guide.pycord.dev"
        authorIcon="/img/logo.png"
        authorName="Pycord Team"
        authorUrl="https://github.com/Pycord-Development/guide"
        thumbnail="/img/logo.png"
        borderColor="#5865F2"
        timestamp="01/09/2021"
        footerIcon="/img/logo.png"
        image="/img/banner-v3.png"
        >
            Pycord Guide is a detailed guide that explains how to use Pycord and all it's features.
            <DiscordEmbedFields slot="fields">
                <DiscordEmbedField fieldTitle="Getting Started">
                    The <a href="/category/getting-started">Getting Started section</a> explains how you can get a brand new bot created and running from scratch.
                </DiscordEmbedField>
                <DiscordEmbedField fieldTitle="Interactions" inline="true">
                    <a href="/interactions">Interactions</a> with Pycord
                </DiscordEmbedField>
                <DiscordEmbedField fieldTitle="Extensions" inline="true">
                    Pycord's various <a href="/category/extensions">Extensions</a>
                </DiscordEmbedField>
                <DiscordEmbedField fieldTitle="Popular Topics" inline="true">
                    <a href="/category/popular-topics">Other Popular Topics</a>
                </DiscordEmbedField>
                <DiscordEmbedField fieldTitle="And More!">
                    We have so much more! Just explore, and you will find everything you need. If you want another page added, open an issue on the <a href="https://github.com/Pycord-Development/guide">GitHub</a>.
                </DiscordEmbedField>
            </DiscordEmbedFields>
            <span slot="footer">Created with 💖 by the Pycord Team & Contributers</span>
        </DiscordEmbed>
    </DiscordMessage>
</DiscordComponent>
```

    embed.add_field(title="Inline Field 2", value="Inline Field 2", inline=True)
    embed.add_field(title="Inline Field 3", value="Inline Field 3", inline=True)
 
    embed.set_footer(text="Footer! No markdown here.") # footers can have icons too
    embed.set_author(name="Pycord Team", icon="https://example.com/link-to-my-image.png")
    embed.set_thumbnail(url="https://example.com/link-to-my-thumbnail.png")
    embed.set_image(url="https://example.com/link-to-my-banner.png")
 
    await ctx.respond("Hello! Here's a cool embed.", embed=embed) # Send the embed with some text
 
bot.run("TOKEN")
```

<DiscordMessages>
    <DiscordMessage author="Guide Bot" avatar="red" bot>
        <div slot="interactions">
            <DiscordInteraction author="Guide Man" avatar="green" command>hello</DiscordInteraction>
        </div>
        <DiscordEmbed
        embedTitle="My Amazing Embed"
        authorIcon="/img/logo.png"
        authorName="Pycord Team"
        authorUrl="https://github.com/Pycord-Development/guide"
        thumbnail="/img/logo.png"
        borderColor="#5865F2"
        footerIcon="/img/logo.png"
        image="/img/banner-v3.png"
        >

>             Embeds are super easy, barely an inconvinience.

Reviewer 2: ```suggestion
            Embeds are super easy, barely an inconvenience.
```

> <DiscordMessages>
>     <DiscordMessage author="Guide Bot" avatar="red" bot>
>         <div slot="interactions">
>             <DiscordInteraction author="Guide Man" avatar="green" command>hello</DiscordInteraction>
>         </div>
>         <DiscordEmbed
>         embedTitle="My Amazing Embed"
>         authorIcon="/img/logo.png"
>         authorName="Pycord Team"
>         authorUrl="https://github.com/Pycord-Development/guide"
>         thumbnail="/img/logo.png"
>         borderColor="#5865F2"
>         footerIcon="/img/logo.png"
>         image="/img/banner-v3.png"
>         >
>             Embeds are super easy, barely an inconvinience.
>             <DiscordEmbedFields slot="fields">
>                 <DiscordEmbedField fieldTitle="A Normal Field">
>                     A really nice field with some information. <strong>The description as well as the fields support markdown!</strong>
>                 </DiscordEmbedField>
>                 <DiscordEmbedField fieldTitle="Inline Field 1" inline="true">Inline Field 1</DiscordEmbedField>
>                 <DiscordEmbedField fieldTitle="Inline Field 2" inline="true">Inline Field 2</DiscordEmbedField>
>                 <DiscordEmbedField fieldTitle="Inline Field 3" inline="true">Inline Field 3</DiscordEmbedField>
>             </DiscordEmbedFields>
>             <span slot="footer">Footer! No markdown here.</span>
>         </DiscordEmbed>
>     </DiscordMessage>
> </DiscordMessages>

Reviewer: ```suggestion
<DiscordComponent>
    <DiscordMessage author="Guide Bot" avatar="red" bot>
        <div slot="interactions">
            <DiscordInteraction author="Guide Man" avatar="green" command>hello</DiscordInteraction>
        </div>
        <DiscordEmbed
        embedTitle="My Amazing Embed"
        authorIcon="/img/logo.png"
        authorName="Pycord Team"
        authorUrl="https://github.com/Pycord-Development/guide"
        thumbnail="/img/logo.png"
        borderColor="#5865F2"
        footerIcon="/img/logo.png"
        image="/img/banner-v3.png"
        >
            Embeds are super easy, barely an inconvinience.
            <DiscordEmbedFields slot="fields">
                <DiscordEmbedField fieldTitle="A Normal Field">
                    A really nice field with some information. <strong>The description as well as the fields support markdown!</strong>
                </DiscordEmbedField>
                <DiscordEmbedField fieldTitle="Inline Field 1" inline="true">Inline Field 1</DiscordEmbedField>
                <DiscordEmbedField fieldTitle="Inline Field 2" inline="true">Inline Field 2</DiscordEmbedField>
                <DiscordEmbedField fieldTitle="Inline Field 3" inline="true">Inline Field 3</DiscordEmbedField>
            </DiscordEmbedFields>
            <span slot="footer">Footer! No markdown here.</span>
        </DiscordEmbed>
    </DiscordMessage>
</DiscordComponent>
```

<br/>

This simple command sends replies to a [slash command](../interactions/application-commands/slash-commands) with an embed.
Let's break it down:
 
```py
embed = discord.Embed(
        title="My Amazing Embed",

>         description="Embeds are super easy, barely an inconvinience.",

Reviewer 2: ```suggestion
        description="Embeds are super easy, barely an inconvenience.",
```

- Image - With the [`set_image`](https://docs.pycord.dev/en/master/api.html#discord.Embed.set_image)
method, you can set an image to sit at the bottom of an embed. This has a single `url` kwarg.
 

> There are a lot more methods and attributes you can use to configure embeds. Here, we just covered the basics. Also remember that all of these values are not nessesary in an embed. An embed may only contain a few of these, for example, only a description, or a title and a description, and so on.

Reviewer 2: ```suggestion
There are a lot more methods and attributes you can use to configure embeds. Here, we just covered the basics. Also remember that all of these values are not necessary in an embed. An embed may only contain a few of these, for example, only a description, or a title and a description, and so on.
```
